x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/search_index.tsx

```diff
@@ -250,18 +239,56 @@ export const SearchIndex: React.FC = () => {
         rightSideItems: getHeaderActions(index),
       }}
     >
-      {isCrawlerIndex(index) && !index.connector ? (
-        <NoConnectorRecord />
-      ) : isCrawlerIndex(index) && (Boolean(errorConnectingMessage) || !config.host) ? (
-        <ErrorStatePrompt />
-      ) : (
-        <>
-          {indexName === index?.name && (
-            <EuiTabbedContent tabs={tabs} selectedTab={selectedTab} onTabClick={onTabClick} />
-          )}
-          {isCrawlerIndex(index) && <CrawlCustomSettingsFlyout />}
-        </>
-      )}
+      <Content
+        index={index}
+        errorConnectingMessage={errorConnectingMessage}
+        config={config}
+        tabs={tabs}
+        tabId={tabId}
+      />
     </EnterpriseSearchContentPageTemplate>
   );
 };
+
+interface ContentProps {
+  config?: ClientConfigType;
+  errorConnectingMessage: string;
+  index?: ElasticsearchViewIndex;
+  tabId?: string;
+  tabs: EuiTabbedContentTab[];
+}
+
+const Content: React.FC<ContentProps> = ({
+  config,
+  errorConnectingMessage,
+  index,
+  tabs,
+  tabId,
+}) => {
+  const selectedTab = useMemo(() => tabs.find((tab) => tab.id === tabId), [tabId]);
+
+  const onTabClick = (tab: EuiTabbedContentTab) => {
```

Comment:
Should we add a `useCallback` to this as well?

Reply:
Good point, fixed in latest

Reply:
Aaannnnd I hit merge before the push went through 🤦